src/unittests/test_group.spec.ts

```diff
@@ -96,30 +96,32 @@ g.test('duplicate_test_name').fn(t => {
 g.test('duplicate_test_params').fn(t => {
   const g = makeTestGroupForUnitTesting(UnitTest);
 
+  g.test('abc')
+    .params([
+      { a: 1 }, //
+      { a: 1 },
+    ])
+    .fn(() => {
+      //
+    });
   t.shouldThrow('Error', () => {
-    g.test('abc')
-      .params([
-        { a: 1 }, //
-        { a: 1 },
-      ])
-      .fn(() => {
-        //
-      });
+    g.checkForDuplicateCases();
   });
 });
 
 g.test('duplicate_test_params,with_different_private_params').fn(t => {
   const g = makeTestGroupForUnitTesting(UnitTest);
 
+  g.test('abc')
+    .params([
+      { a: 1, _b: 1 }, //
+      { a: 1, _b: 2 },
+    ])
+    .fn(() => {
+      //
+    });
   t.shouldThrow('Error', () => {
-    g.test('abc')
-      .params([
-        { a: 1, _b: 1 }, //
-        { a: 1, _b: 2 },
-      ])
-      .fn(() => {
-        //
-      });
+    g.checkForDuplicateCases();
```

Comment:
nit: And add a test that g.checkForDuplicateCases doesn't throw for unparameterized tests

Reply:
bleh, in testing I remembered why it was supposed to be n^2: if params are `{a:1,b:2}, {b:2,a:1}`

Reply:
Oh, I missed that you had switched it to a Set and back. It can't be so bad anyway, there should be a someone reasonable-ish number of params.

Reply:
Yeah, it's fine since now that it's only at build time. It might become a problem eventually, but that can be dealt with then.